server/src/main/java/org/elasticsearch/index/mapper/MappingParser.java

```diff
@@ -22,34 +20,58 @@
     private final Supplier<Mapper.TypeParser.ParserContext> parserContextSupplier;
     private final RootObjectMapper.TypeParser rootObjectTypeParser = new RootObjectMapper.TypeParser();
+    private final Supplier<Map<Class<? extends MetadataFieldMapper>, MetadataFieldMapper>> metadataMappersSupplier;
     private final Map<String, MetadataFieldMapper.TypeParser> rootTypeParsers;
+    private final Function<String, String> documentTypeResolver;
 
-    DocumentMapperParser(IndexSettings indexSettings,
-                         IndexAnalyzers indexAnalyzers,
-                         Function<String, String> documentTypeResolver,
-                         DocumentParser documentParser,
-                         Supplier<Map<Class<? extends MetadataFieldMapper>, MetadataFieldMapper>> metadataMappersSupplier,
-                         Supplier<Mapper.TypeParser.ParserContext> parserContextSupplier,
-                         Map<String, MetadataFieldMapper.TypeParser> metadataMapperParsers) {
-        this.indexSettings = indexSettings;
-        this.indexAnalyzers = indexAnalyzers;
-        this.documentTypeResolver = documentTypeResolver;
-        this.documentParser = documentParser;
-        this.metadataMappersSupplier = metadataMappersSupplier;
+    MappingParser(Supplier<Mapper.TypeParser.ParserContext> parserContextSupplier,
+                  Map<String, MetadataFieldMapper.TypeParser> metadataMapperParsers,
+                  Supplier<Map<Class<? extends MetadataFieldMapper>, MetadataFieldMapper>> metadataMappersSupplier,
+                  Function<String, String> documentTypeResolver) {
         this.parserContextSupplier = parserContextSupplier;
         this.rootTypeParsers = metadataMapperParsers;
+        this.metadataMappersSupplier = metadataMappersSupplier;
+        this.documentTypeResolver = documentTypeResolver;
+    }
+
+    /**
+     * Verify that there are no remaining fields in the provided map that contained mapped fields
+     *
+     * @param fieldName the name of the field that is being parsed
+     * @param fieldNodeMap the map of fields
+     */
+    public static void checkNoRemainingFields(String fieldName, Map<?, ?> fieldNodeMap) {
+        checkNoRemainingFields(fieldNodeMap, "Mapping definition for [" + fieldName + "] has unsupported parameters: ");
+    }
+
+    /**
+     * Verify that there are no remaining fields in the provided map that contained mapped fields
+     *
+     * @param fieldNodeMap the map of fields
+     * @param message the error message to be returned in case the provided map contains one or more fields
+     */
+    public static void checkNoRemainingFields(Map<?, ?> fieldNodeMap, String message) {
+        if (fieldNodeMap.isEmpty() == false) {
+            throw new MapperParsingException(message + getRemainingFields(fieldNodeMap));
+        }
+    }
+
+    private static String getRemainingFields(Map<?, ?> map) {
+        StringBuilder remainingFields = new StringBuilder();
```

Comment:
Does this do anything significantly different to `Objects.toString(map)`?

Reply:
only different parentheses and separator. I am removing this method, hopefully not too many tests will fail as a result.

Reply:
I backed off this change because it  ends up changing ever so slightly an error message which causes having to update tests, and all this is really not the goal for this PR. We can also make such change separately.